src/builtins/filters/array.rs

```diff
@@ -244,45 +244,49 @@ pub fn map(value: &Value, args: &HashMap<String, Value>) -> Result<Value> {
     Ok(to_value(arr).unwrap())
 }
 
-#[inline]
-fn get_index(i: f64, array: &[Value]) -> usize {
-    if i >= 0.0 {
-        i as usize
-    } else {
-        (array.len() as f64 + i) as usize
-    }
-}
-
 /// Slice the array
 /// Use the `start` argument to define where to start (inclusive, default to `0`)
 /// and `end` argument to define where to stop (exclusive, default to the length of the array)
-/// `start` and `end` are 0-indexed
+/// `start` and `end` are 0-indexed and are allowed to be negative indexed
 pub fn slice(value: &Value, args: &HashMap<String, Value>) -> Result<Value> {
     let arr = try_get_value!("slice", "value", Vec<Value>, value);
     if arr.is_empty() {
         return Ok(arr.into());
     }
 
-    let start = match args.get("start") {
-        Some(val) => get_index(try_get_value!("slice", "start", f64, val), &arr),
+    let len = arr.len() as i64;
+
+    let mut start = match args.get("start") {
+        Some(val) => try_get_value!("slice", "start", i64, val),
```

Comment:
sounds like a breaking change if someone was passing a float

Reply:
Float wasn't documented.
I will add a fallback for floats.

Reply:
If we want to keep floats as Index (what seems like a bad Idea, but is understandable for backwards compatibility) we can close this MR